test/module.test.js

```diff
@@ -1173,6 +1173,22 @@ describe('prefix_and_default strategy', () => {
     expect(links.length).toBe(1)
     expect(links[0].getAttribute('href')).toBe('nuxt-app.localhost/')
   })
+
+  test('canonical SEO link includes query params in canonicalQueries', async () => {
+    const html = await get('/?foo="bar"')
+    const dom = getDom(html)
+    const links = dom.querySelectorAll('head link[rel="canonical"]')
+    expect(links.length).toBe(1)
+    expect(links[0].getAttribute('href')).toBe('nuxt-app.localhost/?foo=%22bar%22')
+  })
+
+  test('canonical SEO link does not include query params not in canonicalQueries', async () => {
+    const html = await get('/?bar="baz"')
+    const dom = getDom(html)
+    const links = dom.querySelectorAll('head link[rel="canonical"]')
+    expect(links.length).toBe(1)
+    expect(links[0].getAttribute('href')).toBe('nuxt-app.localhost/')
+  })
```

Comment:
Could you also test this on a page-level? For example:

```js
  test('canonical SEO link include query params in canonicalQueries on page level', async () => {
    const html = await get('/about-us?foo=baz&page=1')
    const dom = getDom(html)
    const links = dom.querySelectorAll('head link[rel="canonical"]')
    expect(links.length).toBe(1)
    expect(links[0].getAttribute('href')).toBe('nuxt-app.localhost/about-us?page=1')
  })
```

and then in `about.vue` return:
```js
return this.$nuxtI18nHead({ addSeoAttributes: true, canonicalQueries: ['page'] })
```

Reply:
Added a page level test 👍